xml/dhcp_vendor_class.xml

```diff
@@ -0,0 +1,215 @@
+    <description>DELL Laser Printer</description>
+    <example hw.model="B2360dn">Mfg=DELL;Typ=Printer;Mod=Dell B2360dn Laser Printer;Ser=</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Dell"/>
+    <param pos="0" name="hw.family" value="Laser Printer"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="hw.product" value="{hw.family} {hw.model}"/>
+    <param pos="0" name="os.vendor" value="Dell"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Mfg=Hewlett Packard;Typ=Printer;Mod=HP (LaserJet 200|LaserJet 400) (?:color |colorMFP |MFP )?(M\d+\S+);Ser=">
+    <description>HP Multifunction Printer</description>
+    <example hw.family="LaserJet 200" hw.model="M276nw">Mfg=Hewlett Packard;Typ=Printer;Mod=HP LaserJet 200 colorMFP M276nw;Ser=CNC8DD4H9V;</example>
+    <example hw.family="LaserJet 400" hw.model="M401dne">Mfg=Hewlett Packard;Typ=Printer;Mod=HP LaserJet 400 M401dne;Ser=PHGFC71816;</example>
+    <example hw.family="LaserJet 400" hw.model="M401dw">Mfg=Hewlett Packard;Typ=Printer;Mod=HP LaserJet 400 M401dw;Ser=PHGGF06370;</example>
+    <example hw.family="LaserJet 400" hw.model="M401n">Mfg=Hewlett Packard;Typ=Printer;Mod=HP LaserJet 400 M401n;Ser=PHGDD02315;</example>
+    <example hw.family="LaserJet 400" hw.model="M425dn">Mfg=Hewlett Packard;Typ=Printer;Mod=HP LaserJet 400 MFP M425dn;Ser=CNF8G78J7W;</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="HP"/>
+    <param pos="1" name="hw.family"/>
+    <param pos="2" name="hw.model"/>
+    <param pos="0" name="hw.product" value="{hw.family} {hw.model}"/>
+    <param pos="0" name="os.vendor" value="HP"/>
+  </fingerprint>
+
+  <fingerprint pattern="^(?:Hewlett-Packard|HP) (OfficeJet|LaserJet|Printer|JetDirect)">
+    <description>HP Printer</description>
+    <example hw.family="LaserJet">Hewlett-Packard LaserJet</example>
+    <example hw.family="OfficeJet">Hewlett-Packard OfficeJet</example>
+    <example hw.family="LaserJet">HP LaserJet</example>
+    <example hw.family="Printer">HP Printer</example>
+    <example>Hewlett-Packard JetDirect</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="HP"/>
+    <param pos="1" name="hw.family"/>
+    <param pos="0" name="os.vendor" value="HP"/>
+  </fingerprint>
+
+  <fingerprint pattern="^Mfg=LEXMARK;Typ=(?:MFP|Printer);Mod=Lexmark (\S+);Ser=">
+    <description>Lexmark Printer</description>
+    <example hw.model="MX410de">Mfg=LEXMARK;Typ=MFP;Mod=Lexmark MX410de;Ser=70155PLM0Y5YD;</example>
+    <example hw.model="MS310dn">Mfg=LEXMARK;Typ=Printer;Mod=Lexmark MS310dn;Ser=451432LM0X7XY;</example>
+    <param pos="0" name="hw.device" value="Printer"/>
+    <param pos="0" name="hw.vendor" value="Lexmark"/>
+    <param pos="1" name="hw.model"/>
+    <param pos="0" name="os.vendor" value="Lexmark"/>
```

Comment:
For devices like printers the `os` is going to be a device or model specific firmware. In those cases go ahead and set `os.device` to `Printer` please.